bundles/org.openhab.binding.magentatv/src/main/java/org/openhab/binding/magentatv/internal/MagentaTVHandler.java

```diff
@@ -0,0 +1,575 @@
+            updateThingProperties(thingConfig.getProperties());
+            handlerFactory.registerDevice(thingConfig.getUDN(), thingConfig.getTerminalID(), thingConfig.getIpAddress(),
+                    this);
+            control.subscribeEventChannel();
+            control.sendPairingRequest();
+            updateThingProperties(thingConfig.getProperties());
+
+            // check for pairing timeout
+            final int iRefresh = ++idRefresh;
+            pairingWatchdogJob = scheduler.schedule(() -> {
+                if (iRefresh == idRefresh) { // Make a best effort to not run multiple deferred refresh
+                    if (thingConfig.getVerificationCode().isEmpty()) {
+                        setOnlineState(ThingStatus.OFFLINE, "Timeout on pairing request!");
+                    }
+                }
+            }, 15, TimeUnit.SECONDS);
+        }
+    }
+
+    /**
+     * If userID is empty and credentials are given the Telekom OAuth service is
+     * used to query the userID
+     *
+     * @throws MagentaTVException
+     */
+    @SuppressWarnings("null")
+    private void authenticateUser() throws MagentaTVException {
+        if (thingConfig.getUserID().isEmpty()) {
+            // run OAuth authentication, this finally provides the userID
+            String userID = control.authenticateUser(thingConfig.getAccountName(), thingConfig.getAccountPassword());
+            thingConfig.setUserID(userID);
+
+            // Update thing configuration (persistent) - remove credentials, add userID
+            Configuration configuration = this.getConfig();
+            configuration.remove(PROPERTY_ACCT_NAME);
+            configuration.remove(PROPERTY_ACCT_PWD);
+            configuration.remove(PROPERTY_USERID);
+            configuration.put(PROPERTY_ACCT_NAME, "***");
+            configuration.put(PROPERTY_ACCT_PWD, "***");
+            configuration.put(PROPERTY_USERID, userID);
+            this.updateConfiguration(configuration);
+            thingConfig.setAccountName("");
+            thingConfig.setAccountPassword("");
+        } else {
+            logger.debug("Skip OAuth, use existing userID");
+        }
+    }
+
+    /**
+     * Update thing status
+     *
+     * @param mode new thing status
+     * @return ON = power on, OFF=power off
+     */
+    public void setOnlineState(ThingStatus newStatus, String errorMessage) {
+        ThingStatus status = this.getThing().getStatus();
+        if (status != newStatus) {
+            if (newStatus == ThingStatus.INITIALIZING) {
+                logger.fatal("Invalid new thing state: ", newStatus.toString());
+            }
+            if (newStatus == ThingStatus.ONLINE) {
+                updateStatus(newStatus);
+                updateState(CHANNEL_POWER, OnOffType.ON);
+            } else {
+                if (!errorMessage.isEmpty()) {
+                    logger.fatal("Communication Error - {0}, switch Thing offline", errorMessage);
```

Comment:
change to logger .debug

Reply:
again, most users check the log if something happens (or not) what they don't expect.

Reply:
see above